packages/contracts/contracts/LenderCommitmentForwarder.sol

```diff
@@ -3,70 +3,124 @@ pragma solidity >=0.8.0 <0.9.0;
+
 contract LenderCommitmentForwarder is TellerV2MarketForwarder {
+
+
+    enum CommitmentCollateralType {
+        ERC20,
+        ERC721,
+        ERC1155,
+        ERC721_ANY_ID, //used only for lender commitments
+        ERC1155_ANY_ID //used only for lender commitments
+    }
+
+    uint256 public constant PRINCIPAL_PER_COLLATERAL_EXPANSION_FACTOR = 10**16;
+
+
     /**
      * @notice Details about a lender's capital commitment.
-     * @param amount Amount of tokens being committed by the lender.
+     * @param maxPrincipal Amount of tokens being committed by the lender. Max amount that can be loaned.
      * @param expiration Expiration time in seconds, when the commitment expires.
      * @param maxDuration Length of time, in seconds that the lender's capital can be lent out for.
-     * @param minAPR Minimum Annual percentage to be applied for loans using the lender's capital.
+     * @param minInterestRate Minimum Annual percentage to be applied for loans using the lender's capital.
+     * @param collateralTokenAddress The address for the token contract that must be used to provide collateral for loans for this commitment.
+     * @param maxPrincipalPerCollateralAmount The amount of principal that can be used for a loan per each unit of collateral, expanded by 10^8. Use zero for no collateral required.
+     * @param collateralTokenType The type of asset of the collateralTokenAddres (ERC20, ERC721, or ERC1155).
+     * @param lender The address of the lender for this commitment.
+     * @param marketId The market id for this commitment.
+     * @param principalTokenAddress The address for the token contract that will be used to provide principal for loans of this commitment.
+     * @param borrower The address of the borrower that is allowed to accept this commitment.  Zero address is wildcard for any address. 
      */
     struct Commitment {
         uint256 maxPrincipal;
         uint32 expiration;
         uint32 maxDuration;
         uint16 minInterestRate;
+        address collateralTokenAddress;
+        uint256 collateralTokenId; 
+        uint256 maxPrincipalPerCollateralAmount; //zero means infinite . This is expressed using 16 decimals [PRINCIPAL_PER_COLLATERAL_EXPANSION_FACTOR] so 1 wei is 10^16 and 1 usdc is 10^22 
+        CommitmentCollateralType collateralTokenType; //erc721, erc1155 or erc20
+        address lender;
+        uint256 marketId;
+        address principalTokenAddress;
+        address borrower;
     }
-
-    modifier onlyMarketOwner(uint256 marketId) {
-        require(_msgSender() == getTellerV2MarketOwner(marketId));
-        _;
-    }
-
+ 
     // Mapping of lender address => market ID => lending token => commitment
     mapping(address => mapping(uint256 => mapping(address => Commitment)))
-        public lenderMarketCommitments;
+        public __lenderMarketCommitments_deprecated;
+
+    // CommitmentId => commitment
+    mapping(uint256 => Commitment) public lenderMarketCommitments;
```

Comment:
given the change in how we create/update commitments based on ID, we can remove the deprecated mapping and replace it with the new one

Reply:
this will mean all existing commitments would be invalidated but should not be a major concern that will break anything

Reply:
i think that would cause storage collision

Reply:
mappings store values in state based on the slot index of the variable & the key values of each nested mapping. the `keccak256` value of all those put together are the actual storage slots used for each value in a mapping
https://docs.soliditylang.org/en/v0.8.17/internals/layout_in_storage.html#mappings-and-dynamic-arrays